cmd/minishift/cmd/start.go

```diff
@@ -416,6 +420,20 @@ func setSubscriptionManagerParameters() {
 	cluster.RegistrationParameters.Password = viper.GetString(startFlags.Password.Name)
 }
 
+func checkRegistrationCredential(host *host.Host) {
+	needRegistration, _ := registration.NeedRegistration(host)
+	if needRegistration {
+		if cluster.RegistrationParameters.Username == "" {
+			username := pkgUtil.ReadInputFromStdin("Username to register to subscription-manager")
```

Comment:
> I have got this feedback in the past for vagrant-registration that when we ask for username and password it is not intuitive which username and password user supposed to enter.

If we want to be specific, we need to bake this into the registration framework. In it's simplest form, one would leave the values in RegistrationParameters nil and let the registrator take care of this. In our RedHatRegistrator in redhead.go. It becomes then the responsibility of the reregistrar to deal with non specified credentials. Works as well for me. Might be even nicer, since it allows you to be more specific in the message you provide. 

To integrate this nicely with the registration framework, the RegistrationParameters could provide a pointer to a function which can be used to interactively receive credentials. This way the solution would not only work for the Red Hat registration. We provide the right function to all registrars in case the credentials are not set via environment or command lines. It is up to the registrar then to either call the function or abort registration.